### ISSUE TITLE
Code Table Request - Change forelimb to limb, add fore as part modfier to all existing forelimb parts and change appendage to limb and remove appendage

### ISSUE THREAD

**Issue by KatherineLAnderson** [ArctosDB/arctos#7670]
### Goal

~~Add limb to the specimen part names code table~~

Change forelimb to limb and add the part modifier fore to existing parts. Change appendage to limb and remove appendage from the code table.

### Context

Limb doesn't exist in the table currently. (Suggested part name "leg" is not consistent with paleo data. Limb is consistent with the paleo literature and what paleontologists would search for to find relevant data.)

If we are going to keep forelimb, then we also need hindlimb and limb (for cases of unknown position).

@falco-rk originally requested limb or appendage and we ended up with appendage there, but now we are going to go with limb.

### Table

https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name

### Proposed Value

~~fore~~limb
~~appendage~~

### Proposed Definition

~~either the forelimb or the hindlimb of a vertebrate~~

A jointed, muscled appendage of a tetrapod vertebrate animal used for weight-bearing, terrestrial locomotion and physical interaction with other objects. Use part modifiers such as fore, hind, anterior, or posterior to specify position. | https://en.wikipedia.org/wiki/Limb_(anatomy)

search terms - appendage, leg, arm, forelimb, hindlimb

### Collection type

paleo, bio

#### Attribute data type

n/a

#### Attribute controlled values

n/a

#### Attribute units

n/a

#### Part preservation attribute affect on "tissueness"

n/a

### Priority

high priority--needed for data entry

### Available for Public View

yes

### Helpful Actions

- [x] Add the issue to the [Code Table Management Project](https://github.com/ArctosDB/arctos/projects/13#card-31628184).

- [ ] Please reach out to anyone who might be affected by this change. Leave a comment or add this to the Committee agenda if you believe more focused conversation is necessary.

@ArctosDB/arctos-code-table-administrators 

## Approval
_All of the following must be checked before this may proceed._

_The [How-To Document](https://handbook.arctosdb.org/how_to/How-To-Manage-Code-Table-Requests.html) should be followed. Pay particular attention to terminology (with emphasis on consistency) and documentation (with emphasis on functionality). **No person should act in multiple roles**; the submitter cannot also serve as a Code Table Administrator, for example._

- [x] Code Table Administrator[1] - check and initial, comment, or thumbs-up to indicate that the request complies with the how-to documentation and has your approval
- [x] Code Table Administrator[2] - check and initial, comment, or thumbs-up to indicate that the request complies with the how-to documentation and has your approval
- [x] DBA - The request is functionally acceptable. The term is not a functional duplicate, and is compatible with existing data and code.
- [x] DBA - Appropriate code or handlers are in place as necessary. (ID_References, Media Relationships, Encumbrances, etc. require particular attention)

## Rejection
_If you believe this request should not proceed, explain why here. Suggest any changes that would make the change acceptable, alternate (usually existing) paths to the same goals, etc._

1. _Can a suitable solution be found here? If not, proceed to (2)_
2. _Can a suitable solution be found by Code Table Committee discussion? If not, proceed to (3)_
3. _Take the discussion to a monthly Arctos Working Group meeting for final resolution._

## Implementation
_Once all of the Approval Checklist is appropriately checked and there are no Rejection comments, or in special circumstances by decree of the Arctos Working Group, the change may be made._

- [ ] Review everything one last time.  Ensure the How-To has been followed. Ensure all checks have been made by appropriate personnel.

- [ ] Add or revise the code table term/definition as described above. Ensure the URL of this Issue is included in the definition. URLs should be included *_as text_*, separated by spaced pipes. Do not include HTML in definitions.

_Close this Issue._

_**DO NOT** modify Arctos Authorities in any way before all points in this Issue have been fully addressed; data loss may result._

## Special Exemptions

_In very specific cases and by prior approval of The Committee, the approval process may be skipped, and implementation requirements may be slightly altered. Please note here if you are proceeding under one of these use cases._

1. _Adding an existing term to additional collection types may proceed immediately and without discussion, but doing so may also subject users to future cleanup efforts. If time allows, please review the term and definition as part of this step._
2. _The Committee may grant special access on particular tables to particular users. This should be exercised with great caution only after several smooth test cases, and generally limited to "taxonomy-like" data such as International Commission on Stratigraphy terminology._

**Comment by dustymc** [comment]
Use https://github.com/ArctosDB/arctos/issues/7143

**Comment by KatherineLAnderson** [comment]
> Use #7143

"Leg"  is better than "appendage". 

As mentioned above-- Suggested part name "leg" is not consistent with paleo data. Limb is consistent with the paleo literature and what paleontologists would search for to find relevant data.

"Forelimb" and "hindlimb" are in the code table--why use "leg" instead of "limb" to refer to something that might be one of those 2 things? This is confusing.

**Comment by Nicole-Ridgwell-NMMNHS** [comment]
Funny that the definition we have for leg contains "Consider using limb" when that is not in the code table. I think the argument for leg is that it can be used for invertebrates as well as tetrapods. Would it be sufficient if limb is added as a search term for leg?

**Comment by KatherineLAnderson** [comment]
> Would it be sufficient if limb is added as a search term for leg?

That would be helpful, and that would solve my concern about paleontologists not knowing to search for leg instead of limb!

Although, I am still struggling with reconciling the forelimb and hindlimb, but no generic limb, especially considering the definition for leg also seems to point out this discrepancy. @Jegelewicz any thoughts here?

**Comment by ebraker** [comment]
As a non-paleo person, I think "leg" sounds odd to record in lieu of "limb" for a fossil.  It seems like we should have part = "limb" with 'fore' and 'hind' specified in part remarks.

**Comment by dustymc** [comment]
>remarks

Or [modifier](https://github.com/ArctosDB/arctos/issues/7693).

**Comment by KatherineLAnderson** [comment]
> As a non-paleo person, I think "leg" sounds odd to record in lieu of "limb" for a fossil. It seems like we should have part = "limb" with 'fore' and 'hind' specified in part remarks.

part name=limb
part modifier=fore; hind

^That would work for me, and I think it makes more sense overall.

**Comment by dustymc** [comment]
>k "leg" sounds odd to record in lieu of "limb"

Possibly best to reopen https://github.com/ArctosDB/arctos/issues/7143#issuecomment-1865129226 (or not, I probably get lost in scattered comments and overloaded issues both...)

**Comment by Jegelewicz** [comment]
A limb is an appendage. So are

antenna [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name#antenna)
arm [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name#arm)
cephalic clasper [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name#cephalic_clasper)
chela [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name#chela)
claw [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name#claw)
fin [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name#fin)
flipper [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name#flipper)
foot [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name#foot)
forelimb [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name#forelimb)
leg [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name#leg)
manus [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name#manus)
pereiopod [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name#pereiopod)
pes [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name#pes)
tail [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name#tail)
toe [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name#toe)
wing [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name#wing)

And maybe other stuff I missed. Looking at the parts code table makes me a little queasy.

**Comment by dustymc** [comment]
>queasy

See also https://github.com/ArctosDB/arctos/issues/7667#issuecomment-2067419047 - let's at least not dig the hole any deeper in search of a way out....

**Comment by KatherineLAnderson** [comment]
> A limb is an appendage. So are
> 
> antenna [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name#antenna) arm [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name#arm) cephalic clasper [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name#cephalic_clasper) chela [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name#chela) claw [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name#claw) fin [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name#fin) flipper [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name#flipper) foot [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name#foot) forelimb [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name#forelimb) leg [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name#leg) manus [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name#manus) pereiopod [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name#pereiopod) pes [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name#pes) tail [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name#tail) toe [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name#toe) wing [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name#wing)
> 
> And maybe other stuff I missed. Looking at the parts code table makes me a little queasy.

If we flatten this to "appendage", at what point is that a wastebasket term? 
Is this another case where we need parts-parts??
(Why am I digging myself a hole.)

**Comment by KatherineLAnderson** [comment]
> > queasy
> 
> See also [#7667 (comment)](https://github.com/ArctosDB/arctos/issues/7667#issuecomment-2067419047) - let's at least not dig the hole any deeper in search of a way out....

read my mind.

**Comment by dustymc** [comment]
>parts-parts?

Yes please! (Or we add all other appendages as parts and figure out how to live with millions of terms, let's just not do a bit of both.)

**Comment by Jegelewicz** [comment]
> Or we add all other appendages as parts and figure out how to live with millions of terms

If they all had appendage in search term, would users still find what they need? Would this make managing the code table someone's full time job? Are we just re-creating some ontology that we should be using (and how would we use it if it existed)?

This is one of the issues facing the entire community (not just Arctos) that we should be working on as a global tool.

**Comment by dustymc** [comment]
>would users still find what they need

I'm going with "no, because the enter-er, looking for 'wing', got tired of scrolling at 'ankle' and deemed it close enough." (Or not, I really don't have anything in mind!)

>managing the code table

I suppose it would depend on the requirements, but I still don't see management as much of an obstacle. USING - for both entry and discovery - are the (visible, maybe only to me?) chokepoints.

>ontology

No, just a big sad list.

>entire community

Yes, we need an adventurous ontologist and a proposal writer.

**Comment by KatherineLAnderson** [comment]
Can we add limb to the parts table, and remove forelimb and hindlimb? Change all existing fore- and hind- to part name=limb, modifier=fore or hind?

Seems to at least fix this (a little bit) until we figure out the bits n pieces/parts-parts discussion.

**Comment by dustymc** [comment]
> fix this 

Still looks like a different way of spelling https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name#appendage to me.

**Comment by Jegelewicz** [comment]
> Still looks like a different way of spelling https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name#appendage to me.

baby steps

This is a step in the right direction, let us move that way.

**Comment by KatherineLAnderson** [comment]
> Still looks like a different way of spelling https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name#appendage to me.

OK but......

> > would users still find what they need
> 
> I'm going with "no"

That's a problem.

I think the request solves the fore/hindlimb problem at minimum (baby steps, right?), but the appendage part name deserves broader discussion.

**Comment by dustymc** [comment]
>step in the right direction

So maybe this is an order of operations problem?

Appendage-users are

```

 guid_prefix | count 
-------------+-------
 OGL:Genomic |   778
 CHAS:Bird   |     2

```

ping


 @droberts49
 @wellerjes
 @falco-rk

How do we do whatever it is that doesn't result in an arbitrary choice between these two (more?) parts.

Of course more.

https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name#arm
https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name#antenna
https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name#claw
https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name#fin
etc etc etc

Having a hard time seeing how adding to that is an improvement, someone help me understand??

**Comment by Jegelewicz** [comment]
> how adding to that is an improvement, someone help me understand??

Nothing is being added? forelimb already exists, we are just simplifying?

**Comment by dustymc** [comment]
>forelimb already exists, we are just simplifying?

<img width="388" alt="Screenshot 2024-04-30 at 12 13 56" src="https://github.com/ArctosDB/arctos/assets/5720791/9a723a9c-07c1-4312-8b38-74af97aa9e3e">

Don't think I read titles....

And for my list: https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_part_name#leg

Ugh.

>simplifying

So there's a migration plan in place?!

**Comment by wellerjes** [comment]
Updated CHAS specimens to remove appendage.

**Comment by falco-rk** [comment]
>Appendage-users are

I'm confused. Do you want me to change my appendages to something else?

I need limb or appendage. When we created appendage, we chose that over limb so it wouldn't be confused with a tree limb. But I don't care which we end up with. I _need_ a generic term for an unknown body appendage where leg (or any of the other parts) is not appropriate or it is not known what appendage was collected. I work with inverts and the can have lots of super specific appendages. I don't need (or want) all those terms. But I do need a generic/non-specific one.

**Comment by Jegelewicz** [comment]
Discussed today at AWG and agreed this is the path to take.

@dustymc @ArctosDB/arctos-code-table-administrators check boxes?

**Comment by dustymc** [comment]
@falco-rk I'd not mind heading another future cleanup project off at the pass, I'll overextend myself and make a part disappear if you'll confirm it's all the same to you....

>check boxes

@Jegelewicz @mkoo please confirm that I've got the go-ahead to update 'forelimb' for

```

 guid_prefix | count 
-------------+-------
 ALMNH:Mamm  |     1
 ASNHC:Mamm  |     1
 UWZM:Mamm   |     1
 UMNH:Mamm   |     1
 OWU:ES      |     4
 DMNS:Mamm   |     1
 UAM:Mamm    |    10
 MSB:Mamm    |    32
 UWBM:Mamm   |    16
```

**Comment by falco-rk** [comment]
I am happy with either limb or appendage.

**Comment by Jegelewicz** [comment]
@falco-rk I don't think this is going to change anything you are using. This issue will have no effect on appendage.

**Comment by dustymc** [comment]
>no effect on appendage

I have every intention of moving OGL's data and deleting appendage; I think they're completely interchangeable, not a fan of multiple ways of saying the same thing...

Alternatively we could move limb-things to appendage.

**Comment by Jegelewicz** [comment]
Possible to get this done now?

@dustymc should I add limb or do you want to manage this whole thing?

**Comment by mkoo** [comment]
sorry missed the updated comments here:
so my understanding and endorsement is for ```limb``` and all other details in attribute! (I also checked the CT admin box with that understanding) Thx!

**Comment by dustymc** [comment]
[temp_part_appendage.csv.zip](https://github.com/ArctosDB/arctos/files/15205268/temp_part_appendage.csv.zip)

[temp_part_forelimb.csv.zip](https://github.com/ArctosDB/arctos/files/15205272/temp_part_forelimb.csv.zip)

done

**Comment by campmlc** [comment]
Just saw this. Limbs are actually derived from appendages, not the other way around. For example, chelicerae. See https://evolution.berkeley.edu/glossary/appendage/#:~:text=Any%20limb%20that%20extends%20from,and%20so%20are%20considered%20appendages.

**Comment by mkoo** [comment]
True. Limbs are a type of appendage-- if Ento or Arthropod collections need a broader term, let's take on appendage then (becuase then we can discuss antennae, etc). AWG meeting settled on limb to include forelimb and hindlimb as attributes. Collections in migration who use an unspecified limb part were included. thx

**Comment by dustymc** [comment]
This has been completed, please open a new issue if there are additional concerns.
